server/py/services/api/initial_data.py

```diff
@@ -52,92 +53,119 @@
 
 
 def init_data(
-    from_scratch: bool = False, perform_migrations_if_needed: bool = False
+    from_scratch: bool = False,
+    perform_migrations_if_needed: bool = False,
 ) -> None:
-    logger.info("Initializing DB data")
+    mlrun.utils.logger.info("Initializing DB data")
 
     alembic_util = None
+    engine = framework.db.sqldb.sql_session.get_engine()
+    url = engine.url
+    if not database_exists(url):
```

Comment:
move functionality to dedicated functions
if X:
  create_db()
else:
  migrate_db()

something like that so it would be easier to read/track the flow